app/code/Magento/Catalog/Model/ProductLink/CollectionProvider.php

```diff
@@ -88,7 +88,7 @@ function (array $itemA, array $itemB): int {
                 $posA = (int)$itemA['position'];
                 $posB = (int)$itemB['position'];
 
-                return $posA <=> $posB;
+                return (($posA - $posB) > 0) ? 1 : -1;  // for PHP 7 and 8 consistency
```

Comment:
hi @ihor-sviziev. in php 8 stable sorting was introduced.  Before, the comparison functions only considered "more than" and "less than or equal to" results, but now they distinguish between "more than" "less than" and "equal to". (https://wiki.php.net/rfc/stable_sorting) There is a test for the function we are discussing, and it seems that a certain order is important here, which is different in php 7 and 8 because of the changes I described. So, in order not to break the existing order, in this change I simulated the behavior from php 7.